### ISSUE TITLE
Vue 3 with vuelidate@next. Needs a way to reset $v while loading another view.

### ISSUE THREAD

**Issue by haoxi911** [vuelidate/vuelidate#765]
I am using `vue-router`, and have some auth forms like below:
```
  {
    path: "/auth",
    children: [
      {
        path: "login",
        component: () => import("@/views/auth/Login.vue")
      },
      {
        path: "register",
        component: () => import("@/views/auth/Register.vue")
      }
    ]
  }
```

I am adding `$v` to `main.ts` like below:
```
import { VuelidatePlugin, Validation } from "@vuelidate/core";

declare module "@vue/runtime-core" {
  interface ComponentCustomProperties {
    $v: Validation;
  }
}

createApp(App)
  .use(router)
  .use(VuelidatePlugin)
  .mount("#app");
```

I am using `vuelidate` and dumped `{{ $v }}` in both `Login.vue` and `Register.vue`.

The issue is, if I navigate to `register` page first and then to `login` page, the `{{ $v }}` instance will bring some validation data from the `register` page to `login` page. For example, we have a `name` field which exists in `register` page only, but I can dump its data in `{{ $v }}` in the `login` page.

I understand `$v` is registered as a global instance, but should we provide a way for us to programmatically cleanup `$v` while navigate to different page via `vue-router`?

**Comment by shentao** [comment]
Hey! Those views should not "talk" to each other. The only way for them to talk is if they are in a parent-child relation (though not directly) as it is using provide/inject underneath. Would you be able to provide a minimal reproduction on codesandbox?

**Comment by emersoncpaz** [comment]
I'm having a similar problem. On first page load vuelidade works fine on login page. But after logout when login page is loaded again seens like model is diconnected from $v object and the validation doesn't work any more.

**Comment by maleta** [comment]
I have same issue. Second component `this.$v` is showing validations from first component and not even loading it's own `validations`.

**Comment by vsorrokin** [comment]
Had the same issue but commit above fixed the problem.

**Comment by clearlyhope** [comment]
I also have the same issue. If a parent component and child component both define validations() the child component's `$v' attribute only shows validations and fields from the parent component. 
Is this behavior intended?

**Comment by shentao** [comment]
This should be fixed with #790
